pkg/storage/unified/resource/kv_test.go

```diff
@@ -0,0 +1,300 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	badger "github.com/dgraph-io/badger/v4"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func setupTestBadgerDB(t *testing.T) *badger.DB {
+	// Create a temporary directory for the test database
+	opts := badger.DefaultOptions("").WithInMemory(true).WithLogger(nil)
+	db, err := badger.Open(opts)
+	require.NoError(t, err)
+	return db
+}
+
+func TestBadgerKV_Get(t *testing.T) {
+	db := setupTestBadgerDB(t)
+	defer func() {
```

Comment:
I'd suggest using `t.Close(...)`, it can be called from `setupTestBadgerDB` to keep main tests simpler.